Add 1.6.0 release notes (#6916)
spiceai/spiceai: docs/release_notes/v1.6/v1.6.0.md

```

See the [MongoDB Data Connector documentation](https://spiceai.org/docs/components/data-connectors/mongodb).

**OpenAI Responses API Support**: Spice OSS now supports the OpenAI Responses API (`/v1/responses`), which is OpenAI's most advanced interface for generating model responses.

You can now make requests to any compatible model using the standard responses endpoint.

Example `curl` request:

```shell
curl http://localhost:8090/v1/responses \
  -H "Content-Type: application/json" \
  -d '{
    "model": "gpt-4.1",
    "input": "Tell me a three sentence bedtime story about Spice AI."
  }'
```

To make requests to the OpenAI Responses API from `spice chat`, use the `--responses` flag.

Example:
```bash
spice chat --responses # Use the `/v1/responses` endpoint for all completions instead of `/v1/chat/completions`
```

Use [tools](https://platform.openai.com/docs/api-reference/responses/create#responses_create-tools) supported by Open AI's Responses API by specifying the `openai_responses_tools` parameter:
```yaml
models:
    - name: test
      from: openai:gpt-4.1
      params:
          openai_api_key: ${ secrets:SPICE_OPENAI_API_KEY }
          tools: sql, list_datasets

>           openai_responses_tools: web_search, local_shell #  'local_shell', 'code_interpreter', or 'web_search'

Reviewer: ```suggestion
          openai_responses_tools: web_search, code_interpreter #  'code_interpreter', or 'web_search'
```
Local shell got scoped out. I forgot to update the release notes when I did that.

```
These OpenAI-specific tools are only available from the `/v1/responses` endpoint. Any other tools specified via the `tools` parameter are available from both the `/v1/chat/completions` and `/v1/responses` endpoints.

> ## Contributors

Reviewer: Does this list need to be updated? It looks the exact same as that for the 1.5.2 release.

Author: Yeah 👍, it is not final/correct - to be updated based on the final change log

**OpenAI Responses API Support**: The OpenAI Responses API (`/v1/responses`) is now supported, which is OpenAI's most advanced interface for generating model responses.

You can now make requests to any responses compatible model using the new `/v1/responses` endpoint.

Example `curl` request:

```bash
curl http://localhost:8090/v1/responses \
  -H "Content-Type: application/json" \
  -d '{
    "model": "gpt-4.1",
    "input": "Tell me a three sentence bedtime story about Spice AI."
  }'
```

To use responses in `spice chat`, use the `--responses` flag.

Example:

```bash
spice chat --responses # Use the `/v1/responses` endpoint for all completions instead of `/v1/chat/completions`
```

Use OpenAI-hosted [tools](https://platform.openai.com/docs/api-reference/responses/create#responses_create-tools) supported by Open AI's Responses API by specifying the `openai_responses_tools` parameter:

Example `spicepod.yml`:

```yaml
models:
  - name: test
    from: openai:gpt-4.1
    params:
      openai_api_key: ${ secrets:SPICE_OPENAI_API_KEY }
      tools: sql, list_datasets

>       openai_responses_tools: web_search, local_shell #  'local_shell', 'code_interpreter', or 'web_search'

Reviewer: ```suggestion
      openai_responses_tools: web_search, code_interpreter #  'code_interpreter' or 'web_search'
```

```

These OpenAI-specific tools are only available from the `/v1/responses` endpoint. Any other tools specified via the `tools` parameter are available from both the `/v1/chat/completions` and `/v1/responses` endpoints.

Learn more in the [OpenAI Model Provider documentation](https://spiceai.org/docs/components/models/openai).

> **OpenAI Embeddings Connector Usage Tier:** The OpenAI Embeddings Connector now supports specifying account usage tier for embeddings requests, improving the performance of generating text embeddings during dataset load and search by increasing concurrent embeddings requests.

Reviewer 2: ```suggestion
**OpenAI Embeddings & Models Connectors Usage Tier:** The OpenAI Embeddings and Models Connectors now supports specifying account usage tier for embeddings and model requests, improving the performance of generating text embeddings or calling models during dataset load and search by increasing concurrent requests.
```

This applies to both embeddings and models

Example `spicepod.yml`:

```yaml
embeddings:
  - from: openai:text-embedding-3-small
    name: openai_embed
    params:
      openai_usage_tier: tier1
```

> By setting the usage tier to the matching usage tier for your OpenAI account, the Embeddings Connector will increase the maximum number of concurrent requests to match the specified tier.

Reviewer 2: ```suggestion
By setting the usage tier to the matching usage tier for your OpenAI account, the Embeddings and Models Connector will increase the maximum number of concurrent requests to match the specified tier.
```

- Added [Live Orders Analytics with Apache Kafka Data Connector](https://github.com/spiceai/cookbook/tree/trunk/kafka/README.md) - Combine real-time data streaming from Kafka with other datasets
- Added [MongoDB Data Connector](https://github.com/spiceai/cookbook/tree/trunk/mongodb/connector/README.md)  - Use MongoDB as a data source with Spice

The [Spice Cookbook](https://spiceai.org/cookbook) includes 77 recipes to help you get started with Spice quickly and easily.

## Upgrading

To upgrade to v1.6.0, use one of the following methods:

**CLI**:

```console
spice upgrade
```

**Homebrew**:

```console
brew upgrade spiceai/spiceai/spice
```

**Docker**:

Pull the `spiceai/spiceai:1.6.0` image:

```console
docker pull spiceai/spiceai:1.6.0
```

For available tags, see [DockerHub](https://hub.docker.com/r/spiceai/spiceai/tags).

**Helm**:

```console
helm repo update
helm upgrade spiceai spiceai/spiceai
```

**AWS Marketplace**:

🎉 Spice is also now available in the [AWS Marketplace](https://aws.amazon.com/marketplace/pp/prodview-jmf6jskjvnq7i)!

## What's Changed

### Dependencies

> - **Rust**: [Upgraded from 1.86.0 to 1.87.0](https://blog.rust-lang.org/2025/05/15/Rust-1.87.0/).

Reviewer 3: @sgrebnov add DataFusion v48.

Author: Added, thank you
>- **DataFusion**: [Upgraded to v48](https://github.com/apache/datafusion/releases/tag/48.0.1)

- docs: Update qa_analytics.csv (#6641) by [@peasee](https://github.com/peasee) in [#6641](https://github.com/spiceai/spiceai/pull/6641)
- Separate token usage (#6619) by [@Advayp](https://github.com/Advayp) in [#6619](https://github.com/spiceai/spiceai/pull/6619)
- Fix typo in release notes (#6634) by [@Advayp](https://github.com/Advayp) in [#6634](https://github.com/spiceai/spiceai/pull/6634)
- Add environment variable for org token (#6633) by [@Advayp](https://github.com/Advayp) in [#6633](https://github.com/spiceai/spiceai/pull/6633)
- CDC: Compute embeddings on ingest (#6612) by [@mach-kernel](https://github.com/mach-kernel) in [#6612](https://github.com/spiceai/spiceai/pull/6612)
- Add view name to view creation errors (#6611) by [@lukekim](https://github.com/lukekim) in [#6611](https://github.com/spiceai/spiceai/pull/6611)
- Add core logic for running MTEB Quora-based vector search tests in Test Operator (#6607) by [@sgrebnov](https://github.com/sgrebnov) in [#6607](https://github.com/spiceai/spiceai/pull/6607)
- Revert "Update generate-openapi.yml (#6584)" (#6620) by [@Jeadie](https://github.com/Jeadie) in [#6620](https://github.com/spiceai/spiceai/pull/6620)

> Warning: No PR found for commit ed9c38774ff7431e4e79dd7d3d86ebccf524766c

Reviewer 3: @sgrebnov fix warning

Author: Fixed, thank you
